app/controllers/tags_controller.rb

```diff
@@ -63,6 +63,55 @@ def destroy
     end
   end
 
+  def paginated_posts
+    @tag = Tag.find params[:id]
+    if @tag.posts
+      @posts = @tag.posts.order(created_at: :desc)
+      @paginated_posts = @posts.paginate(page: params[:page], per_page: 10)
+      respond_to do |format|
+        format.json {
+          render json: {
+            posts: @paginated_posts,
+            page: @paginated_posts.current_page,
+            page_count: @paginated_posts.total_pages
+          }
+        }
+      end
+    end
+  end
+
+  def paginated_citations
+    @tag = Tag.find params[:id]
+    if @tag.posts
+      @posts = @tag.posts
+        .order(created_at: :desc)
+        .paginate(page: params[:posts_page], per_page: 10)
+
+
+      @post_ids = @posts.map{ |p| p.id}
+
+      @citations = Citation.where(:post_id => @post_ids)
```

Comment:
no hash_rocket style please, just use (post_id: @post_ids)

Reply:
also removed hash rocket from the `show` method